checker/src/summaries.rs

```diff
@@ -266,6 +269,37 @@ impl<'a, 'tcx: 'a> PersistentSummaryCache<'a, 'tcx> {
         }
     }
 
+    /// Creates a Sled database at the given directory path, if it does not already exist.
+    /// The initial value of the database contains summaries of standard library functions.
+    /// The code used to create these summaries are mirai/standard_contracts.
+    fn create_default_summary_store_if_needed(
+        summary_store_directory_str: &str,
+    ) -> std::path::PathBuf {
+        use std::env;
+        use std::fs::File;
+        use std::io::Write;
+        use std::path::Path;
+        use tar::Archive;
+
+        let directory_path = Path::new(summary_store_directory_str);
+        let store_path = directory_path.join(".summary_store.sled");
+        if !store_path.exists() && env::var("MIRAI_START_FRESH").is_err() {
```

Comment:
(Out of scope for this PR: Perhaps we should have all of the MIRAI flags documented somewhere?)

Reply:
Good idea.